homeassistant/components/zwave_js/fan.py

```diff
@@ -225,3 +236,111 @@ def zwave_speed_to_percentage(self, zwave_speed: int) -> int:
+        self, config_entry: ConfigEntry, client: ZwaveClient, info: ZwaveDiscoveryInfo
+    ) -> None:
+        """Initialize the thermostat fan."""
+        super().__init__(config_entry, client, info)
+
+        self._fan_mode = self.info.primary_value
+
+        self._fan_off = self.get_zwave_value(
+            THERMOSTAT_FAN_OFF_PROPERTY,
+            CommandClass.THERMOSTAT_FAN_MODE,
+            add_to_watched_value_ids=True,
+        )
+        self._fan_state = self.get_zwave_value(
+            THERMOSTAT_FAN_STATE_PROPERTY,
+            CommandClass.THERMOSTAT_FAN_STATE,
+            add_to_watched_value_ids=True,
+        )
+
+    async def async_turn_on(
+        self,
+        speed: str | None = None,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Turn the device on."""
+        if self._fan_off:
+            await self.info.node.async_set_value(self._fan_off, False)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the device off."""
+        if self._fan_off:
+            await self.info.node.async_set_value(self._fan_off, True)
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if device is on."""
+        if value := get_value_of_zwave_value(self._fan_off) is None:
+            return None
+        return not cast(bool, value)
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., auto, smart, interval, favorite."""
+        value = get_value_of_zwave_value(self._fan_off)
+        if value is None or str(value) not in self._fan_mode.metadata.states:
+            return None
+        return cast(str, self._fan_mode.metadata.states[str(value)])
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+
+        try:
+            new_state = int(
+                next(
+                    state
+                    for state, label in self._fan_mode.metadata.states.items()
+                    if label == preset_mode
+                )
+            )
+        except StopIteration:
+            raise ValueError(f"Received an invalid fan mode: {preset_mode}") from None
+
+        await self.info.node.async_set_value(self._fan_mode, new_state)
+
+    @property
+    def preset_modes(self) -> list[str] | None:
+        """Return a list of available preset modes."""
+        if not self._fan_mode.metadata.states:
+            return None
+        return list(self._fan_mode.metadata.states.values())
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return SUPPORT_PRESET_MODE
+
+    @property
+    def fan_state(self) -> str | None:
+        """Return the current state, Idle, Running, etc."""
+        value = get_value_of_zwave_value(self._fan_state)
+        if (
+            value is None
+            or self._fan_state is None
```

Comment:
I wish I could simplify this further; this is an impossible condition since we've already checked `get_value_of_zwave_value(self._fan_state) is None` on the line above, but I guess the type checker doesn't know that. It's needed since we're accessing `.metadata` on the next line

Reply:
one option is to check all the places where `get_value_of_zwave_value` is used (I think only climate), and if it won't cause an issue, add the logic to convert the value to a state in the helper. Then you can just return `get_value_of_zwave_value`. You could also add a parameter flag to conditionally translate the value into a state to handle both scenarios